extension/src/experiments/data/index.ts

```diff
@@ -41,6 +41,7 @@ export class ExperimentsData extends BaseData<ExpShowOutput> {
   }
 
   public async update(): Promise<void> {
+    void this.updateAvailableBranchesToSelect()
```

Comment:
[F] FWIW moving this in here means that https://github.com/iterative/vscode-dvc/blob/d92aa5291cebaddc39ba7f5155c8c2dda9886557/extension/src/process/manager.ts#L53-L58 takes effect

Reply:
I did the same thing for what I'm currently working on (cleaning up the table state when changing branches). It makes more sense as well.